### PR TITLE
test(dpp): improve coverage for epoch distribution, JSON safe serialization, and address witness

### PR DESCRIPTION
**Pull request by QuantumExplorer**
## Summary

Adds 75 tests across 4 files in rs-dpp covering real logic.

| File | New Tests | What's covered |
|------|-----------|----------------|
| `fee/epoch/distribution.rs` | 16 | Refund calculation, era boundary crossing, multiplier logic, skip verification |
| `serialization/json/safe_integer.rs` | 22 | JS_MAX_SAFE_INTEGER boundaries, Option<i64/u64> paths, string-encoded numbers |
| `serialization/json/safe_integer_map.rs` | 21 | Nested identifier maps, generic maps, boundary values, error paths |
| `address_funds/witness.rs` | 16 | P2PKH/P2SH encode/decode round-trips, MAX_P2SH_SIGNATURES, BorrowDecode |

## Test plan

- [x] `cargo test -p dpp --lib` — 1832 passed
- [x] `cargo fmt --all` clean
- [ ] CI

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Added extensive test coverage for witness data serialization/deserialization with edge cases and error scenarios.
  * Expanded JSON serialization testing for safe-integer handling, including boundary values and round-trip validation.
  * Enhanced test coverage for storage fee refund calculations and map-based serialization across various edge cases.

### CODE REVIEW

**Comment by thepastaclaw** [comment]
### Review Gate

Commit: `effc9204`

- ✅ **Debounce**: 72m ago (need 30m)
- ✅ **CI checks**: builds passed, 0/2 tests passed
- ✅ **CodeRabbit review**: comment found
- ⏳ **Off-peak hours**: peak window (5am-11am PT) — currently 06:22 AM PT Tuesday

- [ ] Run review now (check to override)

File: packages/rs-dpp/src/serialization/json/safe_integer_map.rs [hunk 1]
```diff
@@ -576,4 +576,280 @@ mod tests {
...
+    }
+
+    // --- Error path tests for nested_map ---
+
+    #[test]
+    fn nested_map_invalid_inner_value_type_fails() {
+        let id = Identifier::random();
+        let json =
+            serde_json::json!({"data": {"1": {id.to_string(Encoding::Base58): "not_a_number"}}});
+        let result = serde_json::from_value::<TestNestedMap>(json);
```
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply packages/rs-dpp/src/serialization/json/safe_integer_map.rs:779]: Fair point — the test name is slightly imprecise. The test correctly exercises the error path (invalid string that can't parse as u64), which is the most likely real-world error. The name is close enough to the intent.
